base/float.jl

```diff
@@ -400,9 +400,19 @@ end
 isequal(x::Float16, y::Float16) = fpiseq(x, y)
 isequal(x::Float32, y::Float32) = fpiseq(x, y)
 isequal(x::Float64, y::Float64) = fpiseq(x, y)
-isless( x::Float16, y::Float16) = fpislt(x, y)
-isless( x::Float32, y::Float32) = fpislt(x, y)
-isless( x::Float64, y::Float64) = fpislt(x, y)
+
+# interpret as sign-magnitude integer
+@inline function _fpint(x)
```

Comment:
This probably doesn't make sense for anything but: 
```suggestion
@inline function _fpint(x::T) where T<:IEEEFLoat
```

Reply:
I don't think we need to overspecify the type for this internal function. The code is actually valid for all types that define the appropriate functions, it just interprets data as sign-magnitude integer.